gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcTransportProvider.java

```diff
@@ -0,0 +1,148 @@
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.grpc;
+
+import com.google.api.core.BetaApi;
+import com.google.api.gax.core.BackgroundResource;
+import com.google.api.gax.rpc.TransportProvider;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import io.grpc.ManagedChannel;
+import java.io.IOException;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * A TransportProvider for grpc.
+ *
+ * <p>This class provides instances of {@link GrpcTransport} using the given {@link
+ * ChannelProvider}.
+ *
+ * <p>There are multiple ways to configure the channel that will be used:
+ *
+ * <ol>
+ *   <li>Set ChannelProvider to an instance of InstantiatingChannelProvider, which will create a
+ *       channel when the service API class is created from the settings class. In this case,
+ *       close() should be called on the service API class to shut down the created channel.
+ *   <li>Set ChannelProvider to an instance of FixedChannelProvider, which passes through an
+ *       already-existing ManagedChannel to the API wrapper class. In this case, calling close() on
+ *       the service API class will have no effect on the provided channel.
+ * </ol>
+ */
+@BetaApi
+public class GrpcTransportProvider implements TransportProvider {
+
+  private final ChannelProvider channelProvider;
+
+  /** Constructs an instance of GrpcTransportProvider. */
+  protected GrpcTransportProvider(ChannelProvider channelProvider) {
+    this.channelProvider = Preconditions.checkNotNull(channelProvider);
+  }
+
+  public final ChannelProvider getChannelProvider() {
+    return channelProvider;
+  }
+
+  public String toString() {
+    return MoreObjects.toStringHelper(this).add("channelProvider", channelProvider).toString();
+  }
+
+  @Override
+  public boolean needsExecutor() {
+    return channelProvider.needsExecutor();
+  }
+
+  @Override
+  public GrpcTransport getTransport() throws IOException {
```

Comment:
If the transport provider is a fixed transport provider, then no creation will be performed. I think the "get" terminology is used in cases where it is not known whether an existing or new instance will be returned (e.g. with singletons, the function name is typically `getInstance`).